controllers/auth.js

```diff
@@ -1,15 +1,19 @@
 import User from "../models/User.js";
 import bcrypt from "bcryptjs";
 import { createToken, verifyToken } from "../utils/jwtHelper.js";
+const Joi = require('joi')
 
 export const loginHandler = async (req, res) => {
   try {
     const { email, password } = req.body;
 
-    if (!email || !password) {
-      return res
-        .status(400)
-        .json({ message: "Email and password are required" });
+    const schema = Joi.object().keys({
+      email: Joi.string().email().required(),
+      password: Joi.string().min(8).required(),
+    });
+    const { error } = schema.validate({ email, password });
```

Comment:
Apply this validation as middleware on login route.
Example: https://github.com/MLSAKIIT/devblogs-backend/blob/96a3ef9b4124bfbe9d58ef2b9dacf101d5ad1613/routes/auth.js#L11